packages/core/src/validate/run.ts

```diff
@@ -206,27 +225,159 @@ function* getConstructorErrors(contractDef: ContractDefinition, decodeSrc: SrcDe
   }
 }
 
-function* getOpcodeErrors(contractDef: ContractDefinition, decodeSrc: SrcDecoder): Generator<ValidationErrorOpcode> {
-  for (const fnCall of findAll('FunctionCall', contractDef, node => skipCheck('delegatecall', node))) {
+function* getOpcodeErrors(
+  contractOrFunctionDef: ContractDefinition,
+  deref: ASTDereferencer,
+  decodeSrc: SrcDecoder,
+): Generator<ValidationErrorOpcode> {
+  yield* getContractOpcodeErrors(
+    contractOrFunctionDef,
+    deref,
+    decodeSrc,
+    {
+      kind: 'delegatecall',
+      pattern: /^t_function_baredelegatecall_/,
+    },
```

Comment:
I wonder if this function would look better if we define these description objects in variables at the top level. Along with the default value for the visited set, the result could be:

```typescript
  yield* getContractOpcodeErrors(contractOrFunctionDef, deref, decodeSrc, opcodes.delegatecall, false);
  yield* getContractOpcodeErrors(contractOrFunctionDef, deref, decodeSrc, opcodes.selfdestruct, false);
```